get_info.php

```diff
@@ -1 +1,308 @@
+    }
+}
+
+/**
+ * Returns true if user_id is a list of users.
+ * @param user_id Value of user_id parameters
+ * @return TRUE if user_id is a list
+ */
+function get_list_of_surveys(
+    $role,
+    $list_of_survey,
+    $bind_variables,
+    $is_instance
+) {
+    //dealing with optional params term and course_code
+    $course = false;
+    if (isset($bind_variables[":course_code"])) {
+        $course = true;
+    }
+    $term = false;
+    if (isset($bind_variables[":term"])) {
+        $term = true;
+    }
+    //get list of survey_choices from a user;
+    $survey_choices = get_query_result(
+        gen_query_surveys($role, $list_of_survey, $is_instance, $term, $course),
+        $bind_variables
+    );
+    $list_of_quesitons = get_query_result(gen_query_questions(), []);
+    if (sizeof($survey_choices) <= 0) {
+        $result = null;
+    } elseif ($list_of_survey) {
+        $result = [];
+        //specific survey data
+        for ($i = 0; $i < sizeof($survey_choices); $i++) {
+            $survey_data = $survey_choices[$i];
+            $survey_data["questions"] = [];
+            if (!isset($survey_data["number_locked_by_course"])) {
+                $survey_data["number_locked_by_course"] = null;
+            }
+            $responses = [];
+            if ($is_instance) {
+                $responses = get_query_result(gen_query_survey_responses(), [
+                    ":survey_id" => $bind_variables[":survey_id"]
+                ]);
+            }
+            for ($j = 1; $j <= 6; $j++) {
+                $question = $list_of_quesitons[$survey_data["choice" . $j] - 1];
+                $question["responses"] = isset(
+                    $responses[$survey_data["choice" . $j] - 1]
+                )
+                    ? $responses[$survey_data["choice" . $j] - 1]["answers"]
+                    : null;
+                $question['position'] = $j;
+                array_push($survey_data["questions"], $question);
+                unset($survey_data["choice" . $j]);
+            }
+            array_push($result, $survey_data);
+        }
+    } else {
+        //list of surveys
+        $result = $survey_choices;
+    }
+    if(sizeof($result) == 1){
+        $result = $result[0];
+    }
+    $survey_package = array('TYPE' => "survey_package", 'DATA' => $result);
+    return $survey_package;
+}
+/**
+ * Returns true if user_id is a list of users.
+ * @param user_id Value of user_id parameters
+ * @return TRUE if user_id is a list
+ */
+function is_list_of_users($user_id)
+{
+    $user_id_list = explode(",", $user_id);
+
+    $list_of_users = false;
+    if (sizeof($user_id_list) > 1) {
+        $list_of_users = true;
+    }
+    return $list_of_users;
+}
+
+/**
+ * This functions sets parameters for gen_query_course_associations based on
+ * what parameter. Function returns the result of gen_query_course_associations.
+ *
+ * @param num The HTTP status code
+ * @return array containing the HTTP status of request
```

Comment:
parameters and return is incorrect. Should be
parameters an associative array containing parameter and values
returns CourseAssociations object according to the UML diagram